src/settings/lib/RuleEditor/RulesEditor.js

```diff
@@ -292,130 +226,159 @@ class RulesEditor extends React.Component {
 
-    if ( filter === '' ) {
+    if (filter === '') {
       return;
     }
 
     // scan rows with '#'
     const res = [];
-    const re = new RegExp(filter,'i');
+    const re = new RegExp(filter, 'i');
     let found = true;
     const rng = {};
-    this.cm.eachLine((lh) => {
-      // test rule line for filer string... if
-      if(/^\s*#/.test(lh.text)){
-        if(!re.test(lh.text)){
+
+    this.cm.eachLine(lh => {
+      const line = this.cm.getLineNumber(lh);
+
+      // test rule line for filter string... if
+      if (/^\s*#/.test(lh.text)) {
+        if (!re.test(lh.text)) {
           if (found) {
-            rng.start = {line: this.cm.getLineNumber(lh)-1, ch: 0};
+            rng.start = { line: line - 1, ch: 0 };
             found = false;
           }
-        } else {
-          if(!found){
-            rng.end = {line: this.cm.getLineNumber(lh)-1, ch: lh.text.length};
-            res.push(Object.assign({}, rng));
-            found = true;
-          }
+        } else if (!found) {
+          rng.end = { line: line - 1, ch: lh.text.length };
+          res.push(Object.assign({}, rng));
+          found = true;
         }
       }
-      if(this.cm.getLineNumber(lh) == this.cm.lastLine()){
-        if(!found) {
-          rng.end = {line: this.cm.getLineNumber(lh), ch: lh.text.length};
+
+      if (line === this.cm.lastLine()) {
+        if (!found) {
+          rng.end = { line, ch: lh.text.length };
           res.push(Object.assign({}, rng));
         }
       }
     });
 
     // if filter not found, mark rows until next '#'
     res.forEach((rn) => {
-      this.filteredSections.push(this.cm.markText(rn.start, rn.end, { collapsed: true, inclusiveLeft: true, inclusiveRight: true, }));
+      this.filteredSections.push(this.cm.markText(rn.start, rn.end, { collapsed: true, inclusiveLeft: true, inclusiveRight: true }));
     });
-  }
+  };
 
-  clearErrors(){
-    this.errWidgets.forEach((w) => {
-      w.clear();
-    });
-  }
+  clearErrors = () => this.errWidgets.forEach(errWidget => errWidget.clear());
 
   // called before codemirror state is internally updated...
-  handleFocus(focused) {
-    if(!focused){
-      this.editorFocused = false;
+  handleFocus = focused => {
+    this.editorFocused = focused;
+
+    if (!this.editorFocused) {
       // if help is present when editor loses focus, hide it...
-      // except for when it loses focuse due to clicking a help option (sheesh).
-      if (!this.cm.state.focused) {
-        if( this.cm.state.completionActive && this.cm.state.completionActive.widget ) {
-          const w = this.cm.state.completionActive.widget;
-          // this.cm.state.focused = false;
-          w.close();
-        }
+      // except for when it loses focus due to clicking a help option
+      const {
+        focused: stateFocused,
+        completionActive,
+      } = this.cm.state;
+
+      if (!stateFocused && completionActive.widget) {
```

Comment:
It looks like the previous version of the code checks if this.cm.state.completionActive is not undefined. I would suggest to return this checking.

Reply:
Great spot, thanks.